easybuild/framework/easyblock.py

```diff
@@ -1306,6 +1306,7 @@ def prepare_step(self):
         """
         self.cfg['unwanted_env_vars'] = env.unset_env_vars(self.cfg['unwanted_env_vars'])
         self.toolchain.prepare(self.cfg['onlytcmod'])
+        self.modules_tool.load(self.cfg['system_modules']) 
```

Comment:
I think the `system_modules` should be loaded *first*, before anything else...

Would that break anything as you're using them now?